### PR TITLE
Modify alert group list index

### PR DESCRIPTION
**Pull request by vstpme**
# What this PR does

Modifies the database index used by the alert group list page to accommodate ordering by [`-started_at`](https://github.com/grafana/oncall/pull/3502/files#diff-68d8b0b2e9b7affe54e5950215e69df4afcc6f7f5dd7eeceb810afda0dd5e1d3R85)

## Which issue(s) this PR fixes

Related to https://github.com/grafana/support-escalations/issues/8567

## Checklist

- [x] Unit, integration, and e2e (if applicable) tests updated
- [x] Documentation added (or `pr:no public docs` PR label added if not required)
- [x] `CHANGELOG.md` updated (or `pr:no changelog` PR label added if not required)

### CODE REVIEW

File: engine/apps/alerts/models/alert_group.py [hunk 1]
```diff
@@ -452,7 +452,10 @@ class Meta:
             "is_open_for_grouping",
         ]
         indexes = [
-            models.Index(fields=["channel_id", "resolved", "acknowledged", "silenced", "root_alert_group_id"]),
+            models.Index(
+                fields=["channel_id", "resolved", "acknowledged", "silenced", "root_alert_group_id", "started_at"],
+                name="alert_group_list_index",
+            ),
```
Comment (vadimkerr) [inline engine/apps/alerts/models/alert_group.py:458]: This is the main change of the PR, adding `started_at` at the end of the index

File: engine/common/api_helpers/paginators.py [hunk 2]
```diff
@@ -80,6 +80,6 @@ class FifteenPageSizePaginator(PathPrefixedPagePagination):
     page_size = 15
 
 
-class TwentyFiveCursorPaginator(PathPrefixedCursorPagination):
+class AlertGroupCursorPaginator(PathPrefixedCursorPagination):
```
Comment (vadimkerr) [inline engine/common/api_helpers/paginators.py:83]: Renaming for clarity, this paginator is only used by the alert group endpoint

**Review by matiasb — APPROVED** [review]
LGTM

File: engine/apps/alerts/migrations/0042_remove_alertgroup_alerts_aler_channel_81aeec_idx_and_more.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    dependencies = [
+        ('alerts', '0041_alertreceivechannel_unique_direct_paging_integration_per_team'),
+    ]
+
+    operations = [
+        migrations.AddIndex(
+            model_name='alertgroup',
+            index=models.Index(fields=['channel_id', 'resolved', 'acknowledged', 'silenced', 'root_alert_group_id', 'started_at'], name='alert_group_list_index'),
+        ),
+        migrations.RemoveIndex(
```
[the remark this answers is not in the document]
Comment (vadimkerr) [reply engine/apps/alerts/migrations/0042_remove_alertgroup_alerts_aler_channel_81aeec_idx_and_more.py:17]: yes, here's the `sqlmigrate` output:

```
--
-- Create index alert_group_list_index on field(s) channel_id, resolved, acknowledged, silenced, root_alert_group_id, started_at of model alertgroup
--
CREATE INDEX `alert_group_list_index` ON `alerts_alertgroup` (`channel_id`, `resolved`, `acknowledged`, `silenced`, `root_alert_group_id`, `started_at`);
--
-- Remove index alerts_aler_channel_81aeec_idx from alertgroup
--
DROP INDEX `alerts_aler_channel_81aeec_idx` ON `alerts_alertgroup`;
```

**Pull request merged**